REPL/latex_symbols: define commands for « and »

SeSodesa opened the pull request:
- The commands are `\guillemotleft` and `\guillemotright`, respectively.
- Aims to close #50397.

SeSodesa committed:
REPL/latex_symbols: define commands for « and »

- The commands are \guildoubleft and \guildoubright, respectively.

stevengj commented:
I guess these are trying to follow the convention set by the existing `\guilsinglleft` and `\guilsinglright` abbreviations for `‹` and `›`, respectively.  Those abbreviations seem to come from the [`csquotes` LaTeX package](https://ctan.org/pkg/csquotes?lang=en).   However `csquotes` defines `\guillemotleft` and `\guillemotright` for `«` and `»`, respectively.   I feel like we should be consistent with the `csquotes` names if we are going to add these.

(Alternatively, the `babel` package uses `\flqq` and `\frqq` for the French/angled double quotes, and `\flq` and `\frq` for the corresponding single quotes.  These are more terse, but I'm not sure that is a big consideration in Julia since none of these are valid identifier characters so they are probably not too common in Julia code.)

SeSodesa commented:
I'm perfectly cool with `\guillemotleft` and `\guillemotright` for `«` and `»`. I'll make the edit right away, if someone does not beat me to it.

SeSodesa committed:
REPL/latex_symbols: change commands for « and »

- The new commands are \guillemotleft and \guillemotright, respectively.

- These commands are better in line with the corresponding commands defined in
  the LaTeΧ package csquotes.

stevengj commented on stdlib/REPL/src/latex_symbols.jl:
```diff
@@ -434,6 +434,8 @@ const latex_symbols = Dict(
     "\\backprime" => "‵",
     "\\guilsinglleft" => "‹",
```
```suggestion
    "\\guilsinglleft" => "‹", # note: \guil* quote names follow the LaTeX csquotes package
```

stevengj committed:
Update stdlib/REPL/src/latex_symbols.jl

stevengj commented:
Marking as "merge me" since this change seems really uncontroversial — we have the single-quote completions, so we should have the corresponding double-quote completions.
